### PR TITLE
DynaWaltz: Line model

### PR DESCRIPTION
**Pull request by dimbdr**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Introduce Line model

### CODE REVIEW

**Review by flo-dup — CHANGES_REQUESTED** [review]
Similarly to the bus model, could you also add a LineGroovyExtension in dynawaltz-dsl

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/automatons/CurrentLimitAutomaton.java [hunk 1]
```diff
@@ -45,8 +45,7 @@ public List<Model> getModelsConnectedTo(DynaWaltzContext context) {
         if (line == null) {
             throw new PowsyblException("Unknown line static id: " + lineStaticId);
         }
-        String connectedStaticId = line.getTerminal(side).getBusBreakerView().getConnectableBus().getId();
-        BlackBoxModel connectedBbm = context.getStaticIdBlackBoxModelMap().get(connectedStaticId);
+        BlackBoxModel connectedBbm = context.getStaticIdBlackBoxModelMap().get(line.getId());
```
Comment (flo-dup) [inline dynawaltz/src/main/java/com/powsybl/dynawaltz/models/automatons/CurrentLimitAutomaton.java:48]: :disappointed: well spotted!

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Override
+    public String getLib() {
+        return "Line";
+    }
+
+    @Override
+    public List<VarMapping> getVarsMapping() {
+        return Collections.emptyList();
```
[the remark this answers is not in the document]
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java:43]: Honestly I don't know.
I made this development according to this example : https://github.com/dynawo/dynawo/blob/master/nrt/data/SMIB/SMIB_BasicTestCases/SMIB_1_StepPm_IIDM/SMIB.dyd

I didn't find any dyd file that describes a line with macroconnectors. So for now, I just assumed there's no varsMapping.

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Override
+    public void write(XMLStreamWriter writer, DynaWaltzContext context) throws XMLStreamException {
+        writer.writeStartElement(DYN_URI, "blackBoxModel");
```
[the remark this answers is not in the document]
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java:48]: Indeed, but I might be wrong on this.

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+public class StandardLine extends AbstractBlackBoxModel implements LineModel {
+
+    private final String id;
+    private final String sidePostfix;
+    private final AtomicInteger atomicInteger = new AtomicInteger();
+
+    public StandardLine(String dynamicModelId, String statidId, String parameterSetId, Branch.Side side) {
+        super(dynamicModelId, statidId, parameterSetId);
+        this.sidePostfix = LineModel.getSuffix(side);
+        this.id = "line" + atomicInteger.incrementAndGet();
```
[the remark this answers is not in the document]
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java]: Well, it's useless. I removed this id in https://github.com/powsybl/powsybl-dynawo/pull/147/commits/419f42f92defeeaa7be7e1303592ecc1ff4b5b50

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/automatons/CurrentLimitAutomaton.java [hunk 5]
```diff
@@ -60,12 +59,24 @@ public List<VarConnection> getVarConnectionsWith(Model connected) {
...
-                new VarConnection("currentLimitAutomaton_IMonitored", connectedLineModel.getIVarName()),
-                new VarConnection("currentLimitAutomaton_order", connectedLineModel.getStateVarName()),
-                new VarConnection("currentLimitAutomaton_AutomatonExists", connectedLineModel.getDesactivateCurrentLimitsVarName())
+                new VarConnection(getMonitored(), connectedLineModel.getIVarName()),
+                new VarConnection(getOrder(), connectedLineModel.getStateVarName()),
+                new VarConnection(getExists(), connectedLineModel.getDesactivateCurrentLimitsVarName())
         );
     }
 
+    public String getMonitored() {
```
[the remark this answers is not in the document]
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/models/automatons/CurrentLimitAutomaton.java]: Done in https://github.com/powsybl/powsybl-dynawo/pull/147/commits/f11e90fee4e0be909b1bb6c7b989366639213639

File: dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/models/lines/LineGroovyExtension.groovy [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import com.powsybl.dynamicsimulation.groovy.DynamicModelGroovyExtension
+import com.powsybl.dynawaltz.DynaWaltzProvider
+import com.powsybl.dynawaltz.models.buses.StandardBus
+import com.powsybl.dynawaltz.models.lines.StandardLine
+import com.powsybl.iidm.network.Branch
+
+import java.util.function.Consumer
+
+/**
+ * An implementation of {@link DynamicModelGroovyExtension} that adds the <pre>Bus</pre> keyword to the DSL
```
Comment (dimbdr) [inline dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/models/lines/LineGroovyExtension.groovy:22]: Oops, it adds the `Line` word, not `Bus`

Comment (dimbdr) [reply dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/models/lines/LineGroovyExtension.groovy]: Fixed

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (flo-dup) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java:39]: Indeed, none so far

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (flo-dup) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/models/lines/StandardLine.java]: Better reuse the write of `AbstractBlackBoxModel`

**Review by flo-dup — APPROVED** [review]

**Pull request merged**